src/pages/files/Share.tsx

```diff
@@ -19,17 +19,14 @@ import { AssetPreview } from './AssetPreview'
 import { AssetSummary } from './AssetSummary'
 import { DownloadActionBar } from './DownloadActionBar'
 
-interface MatchParams {
-  hash: string
-}
-
-export function Share(props: RouteComponentProps<MatchParams>): ReactElement {
+export function Share(): ReactElement {
   const { apiUrl, beeApi } = useContext(SettingsContext)
   const { status } = useContext(BeeContext)
 
-  const reference = props.match.params.hash
+  const { hash } = useParams()
+  const reference = hash!
```

Comment:
I don't think hash can be undefined, but we should consider if it should be validated here.